CONTRIBUTING.md

```diff
@@ -0,0 +1,139 @@
+some pointers and feedback from experienced contributors. It will also ensure that you're not
+duplicating work that someone else is doing.
+
+We use Github issues only for, well, issues (mostly bugs that need to be investigated).
+You can also use Github issues for [feature requests](https://github.com/acinq/eclair/issues?q=is%3Aissue+label%3A"feature+request").
+
+## Recommended Reading
+
+- [Bitcoin Whitepaper](https://bitcoin.org/bitcoin.pdf)
+- [Lightning Network Whitepaper](https://lightning.network/lightning-network-paper.pdf)
+- [Deployable Lightning](https://github.com/ElementsProject/lightning/raw/master/doc/deployable-lightning.pdf)
+- [Understanding the Lightning Network](https://bitcoinmagazine.com/articles/understanding-the-lightning-network-part-building-a-bidirectional-payment-channel-1464710791)
+- [Lightning Network Specification](https://github.com/lightningnetwork/lightning-rfc)
+
+## Recommended Skillset
+
+Eclair uses [Scala](https://www.scala-lang.org/) and [Akka](https://akka.io/).
+Some experience with these technologies is required to contribute.
+There are a lot of good resources online to learn about them.
+
+## Contributor Workflow
+
+To contribute a patch, the workflow is as follows:
+
+1. [Fork repository](https://help.github.com/en/github/getting-started-with-github/fork-a-repo) (only the first time)
+2. Create a topic branch
+3. Add commits
+4. Open a pull request
+
+### Pull Request Philosophy
+
+Pull requests should always be focused. For example, a pull request could add a feature, fix a bug,
+or refactor code; but not a mixture.
+Please also avoid super pull requests which attempt to do too much, are overly large, or overly
+complex as this makes review difficult.
+You should try your best to make reviewers' lives as easy as possible: a lot more time will be
+spent reading your code than the time you spent writing it.
+The quicker your changes are merged to master, the less time you will need to spend rebasing and
+otherwise trying to keep up with the master branch.
+
+When fixing a bug, please start by adding a failing test that reproduces the issue.
+Create a first commit containing that test without the fix: this makes it easy to verify that the
+test correctly failed. You can then fix the bug in additional commits.
+
+When adding a new feature, thought must be given to the long term technical debt and maintenance
+that feature may require after inclusion. Before proposing a new feature that will require
+maintenance, please consider if you are willing to maintain it (including bug fixing).
+
+When addressing pull request comments, we recommend using [fixup commits](https://robots.thoughtbot.com/autosquashing-git-commits).
+The reason for this is two fold: it makes it easier for the reviewer to see what changes have been
+made between versions (since Github doesn't easily show prior versions) and it makes it easier on
+the PR author as they can set it to auto-squash the fixup commits on rebase.
+
```

Comment:
Maybe it goes without saying but it's helpful and _recommended_ to run all the test before opening a pull request and keep the continuous integration build as stable as possible.
```suggestion
It's recommended to take great care in writing test and ensuring the entire test suite has a stable successful outcome, eclair uses continuous integration techniques and having a stable build helps the reviewers with their job.
```